pmlc/rt/vulkan/vulkan_invocation.cc

```diff
@@ -565,13 +565,22 @@ void VulkanInvocation::createPipelineLayout() {
                      "vkCreatePipelineLayout");
 }
 
-void VulkanInvocation::createComputePipeline() {
+void VulkanInvocation::createComputePipeline(uint32_t subgroupSize) {
   if (!curr) {
     throw std::runtime_error{"createComputePipeline: curr is nullptr!"};
   }
 
+  VkPipelineShaderStageRequiredSubgroupSizeCreateInfoEXT subgroupSizeInfo;
+  subgroupSizeInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+  subgroupSizeInfo.requiredSubgroupSize = subgroupSize;
+  subgroupSizeInfo.pNext = NULL;
+
   VkPipelineShaderStageCreateInfo stageInfo = {};
   stageInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+  if (device->isExtensionSupported("VK_EXT_subgroup_size_control"))
```

Comment:
Similar to above, we probably need to skip specifying subgroup size if the value is 1 (or the extension is not supported).